### PR TITLE
OVA should respect normalization in underlying learner

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
Address #300

### CODE REVIEW

File: test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs [hunk 1]
```diff
@@ -739,5 +739,64 @@ public void TestCrossValidationMacroWithNonDefaultNames()
...
+                    Data = scoreOutput.ScoredData
+                };
+                var evalOutput = experiment.Add(evalInput);
+                experiment.Compile();
+                experiment.SetInput(importInput.InputFile, new SimpleFileHandle(env, dataPath, false, false));
+                experiment.Run();
+
+                var data = experiment.GetOutput(evalOutput.OverallMetrics);
+                var schema = data.Schema;
+                var b = schema.TryGetColumnIndex(MultiClassClassifierEvaluator.AccuracyMacro, out int aucCol);
```
Comment (zeahmed) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs]: It seems like `AccuracyMacro` is intended here but integer variable name is `aucCol`. Should it be something like `accCol`?

File: test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs [hunk 2]
```diff
@@ -739,5 +739,64 @@ public void TestCrossValidationMacroWithNonDefaultNames()
...
+                Assert.True(b);
+                using (var cursor = data.GetRowCursor(col => col == aucCol))
+                {
+                    var getter = cursor.GetGetter<double>(aucCol);
+                    b = cursor.MoveNext();
+                    Assert.True(b);
+                    double auc = 0;
```
Comment (zeahmed) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs]: same here!

File: test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs [hunk 3]
```diff
@@ -739,5 +739,64 @@ public void TestCrossValidationMacroWithNonDefaultNames()
                 }
             }
         }
+
+        [Fact]
+        public void TestOvaMacro()
```
Comment (zeahmed) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs]: I see that this PR is for fixing normalization issue. However, this test does not test for normalization anywhere. Is this test specifically for normalization or its just the test for "OVA"?

Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs]: It test it through Accuracy! Without normalization Accuracy macro is only 0.6, and this is how I discover this bug.

File: test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs]: yeah, copy pasted and forgot to change it, thank you!

---
In reply to: [193239585](https://github.com/dotnet/machinelearning/pull/310#discussion_r193239585) [](ancestors = 193239585)

File: test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs (hunk 3 above)
Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs]: Initially I plan just add test for Ova, but metrics were really bad (around 0.6 accuracy) so I start digging and found problem with normalization. So indirectly it tests presence of normalization through accuracy metric.

---
In reply to: [193240523](https://github.com/dotnet/machinelearning/pull/310#discussion_r193240523) [](ancestors = 193240523)

**Pull request merged**